easybuild/tools/github.py

```diff
@@ -190,33 +190,107 @@ class GithubError(Exception):
     pass
 
 
-def fetch_easyconfigs_from_pr(pr, path=None, github_user=None):
-    """Fetch patched easyconfig files for a particular PR."""
-
+def github_api_get_request(request_f, github_user=None, **kwargs):
+    """
+    Helper method, for performing get requests to GitHub API.
+    @param request_f: function that should be called to compose request, providing a RestClient instance
+    @param github_user: GitHub user name (to try and obtain matching GitHub token)
+    @return: tuple with return status and data
+    """
     if github_user is None:
         github_user = build_option('github_user')
+
+    token = fetch_github_token(github_user)
+    url = request_f(RestClient(GITHUB_API_URL, username=github_user, token=token))
+
+    try:
+        status, data = url.get(**kwargs)
+    except socket.gaierror, err:
+        _log.warning("Error occured while performing get request: %s" % err)
+        status, data = 0, None
+
+    _log.debug("get request result for %s: status: %d, data: %s" % (url, status, data))
+    return (status, data)
+
+
+def fetch_latest_commit_sha(repo, account, branch='master'):
+    """
+    Fetch latest SHA1 for a specified repository and branch.
+    @param repo: GitHub repository
+    @param account: GitHub account
+    @param branch: branch to fetch latest SHA1 for
+    @return: latest SHA1
+    """
+    status, data = github_api_get_request(lambda x: x.repos[account][repo].branches)
+    if not status == HTTP_STATUS_OK:
+        tup = (branch, account, repo, status, data)
+        _log.error("Failed to get latest commit sha for branch %s from %s/%s (status: %d %s)" % tup)
+
+    res = None
+    for entry in data:
+        if entry[u'name'] == branch:
+            res = entry['commit']['sha']
```

Comment:
is it intentionally redefining `res`? (this is latest first, and `data` is sorted how?)